### ISSUE TITLE
Brace vertical guide is misaligned for @if, @for, @foreach, etc.

### ISSUE THREAD

**Issue by DamianEdwards** [aspnet/Razor#1257]
They don't line up when the Razor expression begins immediately with the keyword.
## Broken
![image](https://cloud.githubusercontent.com/assets/249088/25365348/ed5b0a60-291c-11e7-8925-7d5c5239b4ca.png)

## Works
![image](https://cloud.githubusercontent.com/assets/249088/25365382/24184e32-291d-11e7-8e67-0a772fd0397f.png)

**Comment by DamianEdwards** [comment]
@NTaylorMullen @rynowak

**Comment by sgoguen** [comment]
As a long time MS user, I don't mean to be dismissive about this issue, but I can tell you I'm not losing sleep over it.

The reason I bring it up is @rynowak just closed an issue related to how the newest Razor broke helpers. My point isn't to gripe.  My point is priorities matter.  I realize you're not asking for a customer's perspective, but if you were:  Issues like this one could languish for a decade and I would sleep like a baby.

I'm here to give you guys a friendly reminder that as an IT guy, I have to validate my choices in technology every year.  It's easier to validate using Microsoft's tools when you guys kick ass on important issues like having a clear, coherent, and well-documented system for creating reusable web components with a low learning overhead like JSX.  It's also important to communicate that you have a good long-term plan, even if it's currently evolving.  Polish and nuanced elegance like impeccably clean auto-formatting is appreciated, but it doesn't help me convince people that Razor is worth the continued investment.

I love the work you guys do and I'm looking forward to all the great things you're going to do.  I've been a loyal user precisely because of the awesome work you guys have done in the past.  In case you don't appreciate that statement, I want to be clear that I am not an inherently loyal person.  You guys earned my loyalty because I could see you had a good vision, took risks, and worked hard to make that vision real.  I believe you will continue that culture of having the audacity to aspire to create awesome things like Razor and beyond.

Feel free to reach out to me if you want the perspective of a regular IT guy like myself.  I'll be more than happy to share what features will make me and my team successful and loyal users.

**Comment by rynowak** [comment]
> I realize you're not asking for a customer's perspective, but if you were: Issues like this one could languish for a decade and I would sleep like a baby.

We may not always take the time to ask, but we're definitely interested. Thanks for your interest and for the kind words.

Yesterday we did some spring cleaning on the Razor repro. Most of the issues we closed have been sitting there since 2014. They mostly fell into two categories, 1) obscure parsing issue that no one has complained about, 2) limitations of the existing templating feature.

Speaking about these issues specifically we do have a strategy, it's very long term, and it's going to build on top of what we did with tag helpers.

1. Invest in our tooling and fundamentals https://github.com/aspnet/Razor/issues/836
2- N: Invest in templating and view composition improvements https://github.com/aspnet/Razor/issues/788

**Comment by NTaylorMullen** [comment]
@DamianEdwards I'm not seeing any guidelines in my cshtml files. Is there a VS option that I'm missing? An extension?

**Comment by DamianEdwards** [comment]
Guessing it's these options:
![image](https://cloud.githubusercontent.com/assets/249088/26078804/85a72228-3975-11e7-8ff7-c613b827fb33.png)

**Comment by NTaylorMullen** [comment]
Was able to repro. So we could potentially fix this but the cost would be a misaligned debugging and possibly formatting experience. Let me try and explain:

The `if` part of the statement dictates where the C# editor places the vertical guide. Given that there's an `@` in front of the `if` it moves the vertical guidelines 1 character forward. We could offset the indentation of the underlying braces of an `@if` statement by 1 character to account for this; however, that would result in the yellow debugging highlights appearing 1 character off from where they should be. It might also impact formatting, we've had similar things impact it in the past. We typically render code behind the scenes at the exact character location as seen in the editor to make debugging and formatting behave as expected.

tl;dr; I don't think it's worth doing.

**Comment by DamianEdwards** [comment]
😞
